Code/Framework/AzToolsFramework/AzToolsFramework/Prefab/Spawnable/InMemorySpawnableAssetContainer.cpp

```diff
@@ -213,63 +212,91 @@ namespace AzToolsFramework::Prefab::PrefabConversionUtils
+
+                    AZ::Data::AssetId assetId = asset->GetId();
+                    AZ::Data::AssetType assetType = asset->GetType();
+
+                    // Always queue load as the next step will stop loading for all PreLoad assets
+                    *asset = AZ::Data::AssetManager::Instance().GetAsset(assetId, assetType, AZ::Data::AssetLoadBehavior::QueueLoad);
+
+                    if (!asset->GetId().IsValid())
+                    {
+                        AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
+                        return false;
+                    }
+
+                    if (loadBehavior == AZ::Data::AssetLoadBehavior::PreLoad)
+                    {
+                        // Only assets that are preloaded need to be waited on.
+                        blockingAssets.push_back(asset);
+                    }
+                    return false;
+                }
+                return true;
+            };
+
+            AZ::SerializeContext::EnumerateInstanceCallContext enumerationContext(
+                callback, nullptr, sc, AZ::SerializeContext::ENUM_ACCESS_FOR_READ, nullptr);
+            sc->EnumerateInstance(&enumerationContext, asset.GetData(), asset.GetType(), nullptr, nullptr);
         }
 
         // For all Preload assets we block until they're ready
         // We do this as a separate pass so that we don't interrupt queuing up all other asset loads
-        for (AZ::Data::Asset<AZ::Data::AssetData>& asset : referencedAssets)
+        for (AZ::Data::Asset<AZ::Data::AssetData>* asset : blockingAssets)
         {
-            if (!asset.GetId().IsValid())
+            asset->BlockUntilLoadComplete();
+            
+            if (asset->IsError())
             {
-                AZ_Error("Prefab", false, "Invalid asset found referenced in scene while entering game mode");
-                continue;
-            }
-
-            const AZ::Data::AssetLoadBehavior loadBehavior = asset.GetAutoLoadBehavior();
+                AZ_Error(
+                    "Prefab", false, "Asset with id %s failed to preload while entering game mode",
+                    asset->GetId().ToString<AZStd::string>().c_str());
 
-            if (loadBehavior != AZ::Data::AssetLoadBehavior::PreLoad)
-            {
                 continue;
             }
 
-            asset.BlockUntilLoadComplete();
-
-            if (asset.IsError())
-            {
-                AZ_Error("Prefab", false, "Asset with id %s failed to preload while entering game mode",
-                    asset.GetId().ToString<AZStd::string>().c_str());
-
-                continue;
-            }
+            // Reset the load behavior back to preload because the async load will have caused the behavior to be set to queued. Some assets
+            // will complain if they're not set to the correct loading behavior.
+            asset->SetAutoLoadBehavior(AZ::Data::AssetLoadBehavior::PreLoad);
```

Comment:
You shouldn't need this as long as you don't overwrite the LoadBehavior in the above code.

Reply:
The new asset was setup with an auto load behavior of `QueueLoad` as that's how the asset was created. Based on your earlier suggestion that was updated, but to still get some overlapping loading going the auto load behavior was explicitly set to `QueueLoad` so it has to be set back here. Practically it shouldn't make a difference after an asset has been loaded, but some code (i.e. in Script Canvas) explicitly checks for the auto load behavior. It doesn't have any side effects as far as I can tell other than keeping track of load flag on the asset.

Reply:
Discussed offline - the QueueLoad() call is enough to get the queueing to happen, the calls to SetAutoLoadBehavior() shouldn't meaningfully affect this loading scenario.  They don't _hurt_ anything, but they're extra code that can be removed.